src/Sentry.NLog/SentryTarget.cs

```diff
@@ -196,6 +208,21 @@ protected override void InitializeTarget()
             }
         }
 
+        /// <inheritdoc />
+        protected override void FlushAsync(AsyncContinuation asyncContinuation)
+        {
+            try
+            {
+                _hubAccessor().FlushAsync(Options.FlushTimeout);
```

Comment:
`FlushAsync` returns a `Task` which is unobserved here. 
This is tricky because NLog's `FlushAsync` returns `void` instead of `Task`.
I assume the contract here is to somehow get the returned task that Sentry's `FlushAsync` return into the `AsyncContinuation` so that the library awaits on its completion from it's background worker or something

Reply:
NLog is old and dusty, and before async Task was created. But as the name says then `asyncContinuation` is meant as continuation.

So you should do like this:

```c#
_hubAccessor().FlushAsync(Options.FlushTimeout).ContinueWith(t => asyncContinuation(t.Exception));
```

This will make NLog correctly wait for the flush to complete.

Reply:
Good catch, @bruno-garcia, and thanks for the help @snakefoot! I'll update this accordingly.